### DISCUSSION TITLE
How to get current wallet which was active on the user session?

### DISCUSSION THREAD

**Discussion by tiennt0212** [WalletConnect/walletconnect-monorepo#2816] [Developer Support]
### Description

I'm using Ethereum Provider.

When I connected to WalletConnect, it's able to connect multiple addresses at the same time. These addresses can be returned by getAccounts() method.

However, these address was mixed and not in order. Therefore, I cannot set the current wallet as `accountsList[0]`. It leads to a problem that my current address on the phone cannot match the selected address on my application.

### Expected result

I wonder does it has any way to get the current wallet address on the Metamask mobile application.

**Comment by mohit-zone** [comment]
facing same issue 
https://github.com/orgs/WalletConnect/discussions/2767

**Comment by finessevanes** [comment]
Try `provider.accounts[0]`

**Reply by tiennt0212** [reply]
Hmm, it doesn't work as I expected. `provider.accounts[0]` likely returns the first element in the result of getAccounts() method.

The first element is not always the activating wallet on the user's device.

**Reply by finessevanes** [reply]
> _likely_ returns the first element in the result of getAccounts() method.

Was this the case?

**Comment by tiennt0212** [comment]
To adapt to this situation, I've used this temporary solution :sweat_smile:  - cache the latest activating address:

Pseudocode:
```
accounts <-- getAccounts()

if accounts.length = 0 :
    skip;
else if accounts.length = 1 :
    store cached_wallet <-- accounts[0];
    CURRENT_WALLET <-- accounts[0]
else :
    get cached_wallet;
    if accounts include cached_wallets:
        CURRENT_WALLET <-- cached_wallets
    else:
        CURRENT_WALLET <-- accounts[0]
  
```
